kafka/src/test/groovy/io/micronaut/configuration/kafka/exceptions/DefaultKafkaListenerExceptionHandlerSpec.groovy

```diff
@@ -0,0 +1,161 @@
+            consumer.committedOffset == 0
+        }
+    }
+
+    void "test commit record on deserialization error"() {
+        given:
+        StringProducer stringProducer = context.getBean(StringProducer)
+        CommitOnDeserializationErrorConsumer consumer = context.getBean(CommitOnDeserializationErrorConsumer)
+
+        when: "A producer sends a message with wrong serialization"
+        stringProducer.sendToCommitTopic("not-a-uuid")
+
+        then: "The message is skipped and committed"
+        conditions.eventually {
+            consumer.currentPosition == 1
+            consumer.committedOffset == 1
+        }
+    }
+
+    void "test do nothing on deserialization error"() {
+        given:
+        StringProducer stringProducer = context.getBean(StringProducer)
+        DoNothingOnDeserializationErrorConsumer consumer = context.getBean(DoNothingOnDeserializationErrorConsumer)
+
+        when: "A producer sends a message with wrong serialization"
+        stringProducer.sendToDoNothingTopic("not-a-uuid")
+
+        then: "The message is neither skipped nor committed"
+        conditions.eventually {
+            consumer.currentPosition == 0
+            consumer.committedOffset == 0
+        }
+    }
+
+    static abstract class AbstractOnDeserializationErrorConsumer implements KafkaListenerExceptionHandler {
+        Long currentPosition = -1
+        Long committedOffset = -1
+        DefaultKafkaListenerExceptionHandler errorHandler = new DefaultKafkaListenerExceptionHandler()
+        String topic
+
+        AbstractOnDeserializationErrorConsumer(String topic) {
+            this.topic = topic
+        }
+
+        @Override
+        void handle(KafkaListenerException exception) {
+            errorHandler.handle(exception)
+            TopicPartition tp = new TopicPartition(topic, 0)
+            currentPosition = exception.kafkaConsumer.position(tp)
+            OffsetAndMetadata committedOffsetAndMetadata = exception.kafkaConsumer.committed(tp)
+            if (committedOffsetAndMetadata != null) {
+                committedOffset = committedOffsetAndMetadata.offset()
+            } else {
+                committedOffset = 0
+            }
+        }
+    }
+
+    @Requires(property = 'spec.name', value = 'DefaultKafkaListenerExceptionHandlerSpec')
+    @KafkaListener(
+            offsetReset = EARLIEST,
+            offsetStrategy = DISABLED,
+            properties = [
+                    @Property(name = ProducerConfig.VALUE_SERIALIZER_CLASS_CONFIG,
+                            value = "org.apache.kafka.common.serialization.UUIDSerializer")
+            ]
+    )
+    static class DefaultBehaviorOnDeserializationErrorConsumer extends AbstractOnDeserializationErrorConsumer {
+        DefaultBehaviorOnDeserializationErrorConsumer() {
+            super(TOPIC_SEEK)
+        }
+
+        @Topic(TOPIC_SEEK)
+        void receive(UUID uuid) {
+        }
+    }
+
+    @KafkaListener(
```

Comment:
this class needs to be annotated with ` @Requires(property = 'spec.name',` as well.